database/poolDataManager.ts

```diff
@@ -0,0 +1,98 @@
+import { DB_CLIENT } from "./dbConfig";
+
+const POOL_TABLE: string = "boba-buddies-pool";
+
+export class PoolDataManager {
+	async getPool(): Promise<Map<string, string>[]> {
+		const params = {
+			TableName: POOL_TABLE
+		};
+
+		const result = await DB_CLIENT.scan(params).promise();
+		return result.Items;
+	}
+
+	async getPoolForUser(user: string): Promise<Map<string, string>[]>  {
+		const getFilteredParams = {
+			TableName: POOL_TABLE,
+			FilterExpression: "(primaryBuddy = :primary) OR (secondaryBuddy = :secondary)",
+			ExpressionAttributeValues: {
+				":primary": user,
+				":secondary": user
+			}
+		}
+		const result = await DB_CLIENT.scan(getFilteredParams).promise();
+		return result.Items;
+	}
+
+	async addPair(slackUser1: string, slackUser2: string): Promise<void> {
+		const params = {
+			TableName: POOL_TABLE,
+			Item: {
+				"primaryBuddy": slackUser1,
+				"secondaryBuddy": slackUser2
+			}
+		}
+
+		return await DB_CLIENT.put(params).promise();
+	}
+
+	async deletePair(pair: Map<string, string>): Promise<void> {
+		const delParams = {
+				TableName: POOL_TABLE,
+				Key: {
+					primaryBuddy: pair.get("primaryBuddy"),
+					secondaryBuddy: pair.get("secondaryBuddy")
+				}
+			}
+		return await DB_CLIENT.delete(delParams).promise();
+	}
+
+	async popPair(user: string): Promise<Map<string, string>> {
+		const userPool = await this.getPoolForUser(user);
+		const randomPair = userPool[Math.floor(Math.random() * userPool.length)];
+		await this.deletePair(new Map(Object.entries(randomPair)));
+
+		return randomPair;
```

Comment:
omfg im dumb and i didn't account for previously popped users. this a bug

Reply:
planning to do: configure this method to take in a list of already paired users and run a filter query into dynamodb